src/endpointRoutes.js

```diff
@@ -0,0 +1,140 @@
+'use strict'
+
+const KoaBodyParser = require('@viweei/koa-body-parser')
+
+const logger = require('./logger')
+const {handleServerError, validateEndpoint} = require('./util')
+const {
+  deleteEndpoint,
+  createEndpoint,
+  updateEndpoint
+} = require('./db/services/endpoints')
+
+const createEndpointRoute = router => {
+  router.post('/endpoints', async (ctx, next) => {
+    const failureMsg = 'Endpoint creation/update failed: '
+
+    try {
+      await KoaBodyParser()(ctx, () => {})
+
+      if (
+        !ctx.request ||
+        !ctx.request.body ||
+        !Object.keys(ctx.request.body).length
+      ) {
+        ctx.status = 400
+        const err = `${failureMsg}Invalid endpoint object`
+        ctx.body = {error: err}
+        logger.error(err)
+        return next()
+      }
```

Comment:
We could get rid of this validation check as mongoose will perform the same thing. If no object is sent through mongoose will complain about the failed validation about missing properties

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/48/commits/8a33c66e55c35491ef334d272040f65147472c45